fancy-weather/src/index.js

```diff
@@ -0,0 +1,426 @@
+  elementNext1DayTemp.innerHTML = `${Math.round(data.weather.nextFirstDayTemp)}&#176`;
+  elementNext2DayTemp.innerHTML = `${Math.round(data.weather.nextSecondDayTemp)}&#176`;
+  elementNext3DayTemp.innerHTML = `${Math.round(data.weather.nextThirdDayTemp)}&#176`;
+  elementLatitude.innerHTML = `${dictionary[currentLanguage].latitude}: ${String(data.geocoding.latitude).split('.')[0]}&deg${String(data.geocoding.latitude).split('.')[1].slice(0, 2)}&#39`;
+  elementLongitude.innerHTML = `${dictionary[currentLanguage].longitude}: ${String(data.geocoding.longitude).split('.')[0]}&deg${String(data.geocoding.longitude).split('.')[1].slice(0, 2)}&#39`;
+}
+
+function loadIcons() {
+  elementCurrentIcon.src = `icons/${data.weather.currentIcon}.svg`;
+  elementNext1DayIcon.src = `icons/${data.weather.nextFirstDayIcon}.svg`;
+  elementNext2DayIcon.src = `icons/${data.weather.nextSecondDayIcon}.svg`;
+  elementNext3DayIcon.src = `icons/${data.weather.nextThirdDayIcon}.svg`;
+}
+
+function loadMap() {
+  if (document.documentElement.clientWidth > verticalScreenWidth) {
+    elementMapContainer.style.height = `${elementNext3DayTemp.getBoundingClientRect().top}px`;
+  }
+  elementMap.replaceChildren();
+  const myMap = new ymaps.Map('geolocation__map', {
+    center: [data.geocoding.latitude, data.geocoding.longitude],
+    zoom: 10,
+    controls: ['zoomControl'],
+  });
+  document.querySelector('.ymaps-2-1-78-copyrights-pane').remove();
+}
+
+function enterSearchValue(event) {
+  if (event.code === 'Enter') {
+    controlWeather();
+  }
+}
+
+function showHideMistakeMessage() {
+  if (elementMistakesTint.hidden === false) {
+    elementMistakesTint.hidden = true;
+    elementTownInput.value = '';
+    elementMistakesText.innerHTML = '';
+  } else {
+    elementMistakesTint.hidden = false;
+    elementMistakesText.innerHTML = `${dictionary[currentLanguage].mistake}`;
+  }
+}
+
+function rotationArrows() {
+  elementRefreshButtonArrows.classList.add('arrows_rotation');
+  setTimeout(() => elementRefreshButtonArrows.classList.remove('arrows_rotation'), 1000);
+}
+
+function Initialization() {
+  loadSaveLanguage();
+  loadSaveScale();
+  const promise = new Promise((resolve) => {
+    resolve(getLocationByIP());
+  });
+  promise
+    .then((data) => getCoordinates(coordinatesIP))
+    .then((data) => loadWeather(data))
+    .then((data) => loadPictures(data))
+    .then((data) => setDateAndTime(data))
+    .then((data) => setCurrentWeatherScale(data))
+    .then((data) => fillDOMContent(data))
+    .then((data) => loadIcons(data))
+    .then((data) => loadMap(data))
+    .catch(() => showHideMistakeMessage());
+}
+
+function controlWeather() {
+  const promise = new Promise((resolve) => {
+    resolve();
+  });
+  promise
+    .then((data) => (elementTownInput.value === '' ? getCoordinates(coordinatesIP) : getCoordinates(elementTownInput.value)))
```

Comment:
`` elementTownInput.value === '' `` => `` !elementTownInput.value `` - it's will be enough